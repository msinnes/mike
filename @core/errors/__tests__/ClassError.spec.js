const ClassError = require('../ClassError');
const BaseError = require('../lib/BaseError');

describe('ClassError', () => {
  it('should be a function', () => {
    expect(ClassError).toBeDefined();
    expect(ClassError).toBeInstanceOf(Function);
  });

  it('should make instances of error', () => {
    expect(new ClassError()).toBeInstanceOf(Error);
    expect(new ClassError()).toBeInstanceOf(BaseError);
  });

  it('should express the correct error state', () => {
    try {
      throw new ClassError('message');
    } catch (e) {
      expect(e.name).toEqual('ClassError');
      expect(e.stack).toMatchSnapshot();
    }
  });
});
