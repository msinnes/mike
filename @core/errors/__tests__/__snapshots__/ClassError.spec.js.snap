// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassError should express the correct error state 1`] = `"ClassError: ClassError: message"`;
